src/states/visualizations/reducer.js

```diff
@@ -48,22 +48,24 @@ const visualizations = (state = defaultState, action) => {
                 $set: {
                   modelId: modelId,
                   specificationId: specificationId,
-                  plotId: plotId,
-                  visualizationId: id
-                }
-              }
+                  //plotId: plotId,
+                  visualizationId: id,
```

Comment:
why did you leave out plotId for this one? Because there could be several plots per visualization?

Reply:
because when visualization is first created, the new plot has not created yet. :)

Reply:
because the plot has not yet created yet when we first created the visualization. :) Maybe it makes more sense to create the plot first before creating a visualization? I could look into changing it.